src/shims/windows/foreign_items.rs

```diff
@@ -119,10 +122,59 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     system_info.ptr,
                     iter::repeat(0u8).take(system_info.layout.size.bytes() as usize),
                 )?;
+                // Set selected fields.
+                let word_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::U16));
+                let dword_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::U32));
+                let usize_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::Usize));
+                let word_layout = this.layout_of(word_ty)?;
+                let dword_layout = this.layout_of(dword_ty)?;
+                let usize_layout = this.layout_of(usize_ty)?;
+
+                // Using `mplace_field` is error-prone, see: https://github.com/rust-lang/miri/issues/2136.
+                // Pointer fields have different sizes on different targets.
+                // To avoid all these issue we calculate the offsets dynamically.
```

Comment:
```suggestion
                // To avoid all these issue we calculate the offsets ourselves.
```
Everything here is dynamic, including your previous approach.

Reply:
Agreed.